infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/AsyncStorageProvisioner.java

```diff
@@ -0,0 +1,383 @@
+      }
+    }
+    return sshPairs;
+  }
+
+  /** Create configmap with public part of SSH key */
+  private void createConfigMapIfNotExist(
+      KubernetesClient oc,
+      String namespace,
+      String configMapName,
+      String userId,
+      OpenShiftEnvironment osEnv)
+      throws InfrastructureException {
+    Resource<ConfigMap, DoneableConfigMap> mapResource =
+        oc.configMaps().inNamespace(namespace).withName(configMapName);
+    if (mapResource.get() != null) { // map already exist
+      return;
+    }
+
+    List<SshPairImpl> sshPairs = getOrCreateSshPairs(userId, osEnv);
+    if (sshPairs == null) {
+      return;
+    }
+    SshPair sshPair = sshPairs.get(0);
+    Map<String, String> sshConfigData = of(AUTHORIZED_KEYS, sshPair.getPublicKey() + "\n");
+    ConfigMap configMap =
+        new ConfigMapBuilder()
+            .withNewMetadata()
+            .withName(configMapName)
+            .withNamespace(namespace)
+            .withLabels(of(CHE_USER_ID_LABEL, userId))
+            .endMetadata()
+            .withData(sshConfigData)
+            .build();
+    oc.configMaps().inNamespace(namespace).create(configMap);
+  }
+
+  /**
+   * Create storage Pod with container with mounted volume for storing project source backups, SSH
+   * key and exposed port for rsync connection
+   */
+  private void createAsyncStoragePodIfNotExist(
+      KubernetesClient oc, String namespace, String configMap, String userId) {
+    PodResource<Pod, DoneablePod> podResource =
+        oc.pods().inNamespace(namespace).withName(ASYNC_STORAGE);
+    if (podResource.get() != null) {
+      return; // pod already exist
+    }
+
+    String containerName = Names.generateName(ASYNC_STORAGE);
+
+    Volume storageVolume =
+        new VolumeBuilder()
+            .withName(STORAGE_VOLUME)
+            .withPersistentVolumeClaim(
+                new PersistentVolumeClaimVolumeSourceBuilder()
+                    .withClaimName(configuredPVCName)
+                    .withReadOnly(false)
+                    .build())
+            .build();
+
+    Volume sshKeyVolume =
+        new VolumeBuilder()
+            .withName(CONFIG_MAP_VOLUME_NAME)
+            .withConfigMap(
+                new ConfigMapVolumeSourceBuilder()
+                    .withName(configMap)
+                    .withDefaultMode(0600)
+                    .build())
+            .build();
+
+    VolumeMount storageVolumeMount =
+        new VolumeMountBuilder()
+            .withMountPath(ASYNC_STORAGE_DATA_PATH)
+            .withName(STORAGE_VOLUME)
+            .withReadOnly(false)
+            .build();
+
+    VolumeMount sshVolumeMount =
+        new VolumeMountBuilder()
+            .withMountPath(SSH_KEY_PATH)
+            .withSubPath(AUTHORIZED_KEYS)
+            .withName(CONFIG_MAP_VOLUME_NAME)
+            .withReadOnly(true)
+            .build();
+
+    Container container =
+        new ContainerBuilder()
+            .withName(containerName)
```

Comment:
do we need to add ` .withImagePullPolicy(imagePullPolicy)`?

Reply:
yep, thx